packages/imperative/src/config/src/ProfileInfo.ts

```diff
@@ -179,6 +179,48 @@ export class ProfileInfo {
         this.mImpLogger = ConfigUtils.initImpUtils(this.mAppName);
     }
 
+    /**
+     * Checks if a JWT token is used for authenticating the given profile name. If so, it will decode the token to determine whether it has expired.
+     *
+     * @param {string | IProfileLoaded} profile - The name of the profile or the profile object to check the JWT token for
+     * @returns {boolean} Whether the token has expired for the given profile. Returns `false` if a token value is not set or the token type is LTPA2.
+     */
+    public hasTokenExpiredForProfile(profile: string | IProfileLoaded): boolean {
+        const profName = typeof profile === "string" ? profile : profile.name;
+        const profAttrs = this.getAllProfiles().find((prof) => prof.profName === profName);
+        const knownProps = this.mergeArgsForProfile(profAttrs, { getSecureVals: true }).knownArgs;
+        const tokenValueProp = knownProps.find((arg) => arg.argName === "tokenValue" && arg.argValue != null);
+
+        // Ignore if tokenValue is not a prop
+        if (tokenValueProp == null) {
+            return false;
+        }
+
+        const tokenTypeProp = knownProps.find((arg) => arg.argName === "tokenType");
+        // Cannot decode LTPA tokens without private key
+        if (tokenTypeProp?.argValue == "LtpaToken2") {
+            return false;
+        }
+
+        const fullToken = tokenValueProp.argValue.toString();
+        // JWT format: [header].[payload].[signature]
+        const tokenParts = fullToken.split(".");
+        try {
+            const payloadJson = JSON.parse(Buffer.from(tokenParts[1], "base64url").toString("utf8"));
+            if ("exp" in payloadJson) {
+                // The expire time is stored in seconds since UNIX epoch.
+                // The Date constructor expects a timestamp in milliseconds.
+                const msPerSec = 1000;
+                const expireDate = new Date(payloadJson.exp * msPerSec);
+                return expireDate < new Date();
+            }
+        } catch (err) {
+            return false;
+        }
+
+        return false;
```

Comment:
Thanks for adding this to the ProfileInfo API! As the ProfileInfo API is used only by extenders like ZE, I'm curious if we'd want to move this part of the method to make it usable by CLI itself?

Perhaps could have a static `isTokenExpired` method on the `ConfigUtils` or `RestStandaloneUtils` classes, that only checks if the token is expired and doesn't handle loading it from config.

Reply:
Good call - I added a function to `ConfigUtils` called `hasTokenExpired` and updated the `ProfileInfo` function to use it.
Also added an entry for the new function in the changelog 👍